### PR TITLE
tests/qdsa: Move from unittests to regular tests

### PR DESCRIPTION
**Pull request by bergzand**
And another!

### Contribution description

similar to #10183, this is another large package that only increases the size of the unittests.

### Testing procedure

Run the `tests/pkg_qdsa` test.

### Issues/PRs references

Part of https://github.com/RIOT-OS/RIOT/issues/10187

### CODE REVIEW

**Comment by kaspar030** [comment]
LGTM.

**Comment by bergzand** [comment]
Adjusted blacklist and insufficient memory list based on Murdock results.

**Comment by bergzand** [comment]
Test takes around 66 seconds on a samr21-xpro:
```
2018-10-17 22:49:44,610 - INFO # main(): This is RIOT! (Version: 2018.10-devel-1213-g386c9-dystopia-pr/tests/libcose_move)
2018-10-17 22:50:50,990 - INFO # ...
2018-10-17 22:50:50,991 - INFO # OK (3 tests)
```

Also here the question whether we want to have this run on the CI?

**Comment by bergzand** [comment]
> This one takes only 7.6 seconds on `samr21-xpro` I think you mixed with the `libcose` test according to the output :)

You're right, mixed up a few numbers. As the PR is now, the test is run on `all`.

File: tests/pkg_qdsa/tests/01-run.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+import sys
+from testrunner import run
+
+
+def testfunc(child):
+    child.expect('OK \(\d+ tests\)', timeout=120)
```
[the remark this answers is not in the document]
Comment (cladmi) [reply tests/pkg_qdsa/tests/01-run.py]: To be more precise, the `7.6` seconds I mentioned in the main review is the total time counted by `time`.
But the time actually spend in the test is `~3` seconds.

[the remark this answers is not in the document]
Comment (bergzand) [reply tests/pkg_qdsa/tests/01-run.py]: Ack, sloppy copy-paste code after moving too many package tests :)

[the remark this answers is not in the document]
Comment (bergzand) [reply tests/pkg_qdsa/tests/01-run.py]: removed!

File: tests/pkg_qdsa/Makefile [hunk 2]
```diff
@@ -0,0 +1,16 @@
+include ../Makefile.tests_common
+
+TEST_ON_CI_WHITELIST += all
+
+# qDSA is not 16 bit compatible
+BOARD_BLACKLIST := chronos msb-430 msb-430h telosb wsn430-v1_3b wsn430-v1_4 z1
+
+BOARD_INSUFFICIENT_MEMORY := arduino-duemilanove arduino-uno nucleo-f031k6
+
+CFLAGS += -DTHREAD_STACKSIZE_MAIN=\(4*THREAD_STACKSIZE_DEFAULT\)
```
Comment (cladmi) [inline tests/pkg_qdsa/Makefile]: As the issue also happened in `test/tweetnacl` I think you should keep the increased stack size for avrboards

```
AVR_BOARDS := arduino-duemilanove \
               arduino-mega2560 \
               arduino-uno \
               waspmote-pro \
#
```
```
CFLAGS += -DTHREAD_STACKSIZE_MAIN=\(4*THREAD_STACKSIZE_DEFAULT+THREAD_EXTRA_STACKSIZE_PRINTF\)
```

This way, we could merge this without re-checking them for now.

Comment (bergzand) [reply tests/pkg_qdsa/Makefile]: Is it also okay if I increase the `THREAD_STACKSIZE_DEFAULT` multiplier by 1? It doesn't really make sense to increase the stack size by `THREAD_EXTRA_STACKSIZE_PRINTF` if there are no printf statements in the code.

Comment (cladmi) [reply tests/pkg_qdsa/Makefile]: Then it should be re-tested on `avr` which I cannot do today.
By only moving them without changing the "absurd" size, it is easy to merge. 
Fixing the stack size can be another task.

Comment (bergzand) [reply tests/pkg_qdsa/Makefile]: There is only one platform where increasing it to `5*THREAD_STACKSIZE_DEFAULT` would result in a lower total stack size compared to `4*THREAD_STACKSIZE_DEFAULT+THREAD_EXTRA_STACKSIZE_PRINTF`: the lpc2387. For all other platforms, the `THREAD_STACKSIZE_DEFAULT` is larger than the `THREAD_EXTRA_STACKSIZE_PRINTF`. 

I don't mind if you want to retest on the `avr` platforms anyway, I'm not in that much of a hurry with this.

Comment (cladmi) [reply tests/pkg_qdsa/Makefile]: Oh, I did not check the exact values for this. Thank you for researching it.
If 5* is bigger no need to retest for them.

So I like to keep the 5 * here as you updated and please explain it in the commit message (also that the lpc2387 is now smaller and not re-tested).
Putting your last comment directly is good for me.

Comment (cladmi) [reply tests/pkg_qdsa/Makefile]: This concern and solving justification also applies for the other `LARGE_STACK_TESTS` tests you migrated.

**Comment by cladmi** [comment]
Agreed with the timeout, only need to update to `5*THREAD_STACKSIZE_DEFAULT` and put it in the commit message.
You can directly squash when changing these.

I tested with `iotlab-m3` and `samr21-xpro` and the timeout is good for them. No corrupt stack.

**Comment by bergzand** [comment]
@cladmi squashed and reworded the commit message. Please check whether it contains al the requested information. Thanks for the thorough testing of this PR (and all the related PRs)

**Comment by cladmi** [comment]
The commit says `5*` but the change is still `4*THREAD_STACKSIZE_DEFAULT`.

However I now tested with `arduino-mega2560` and it works without warning message. A `ps` gives me this for `main` thread:

```
Stack ( used)
   1024 (  734)
```

From the other avr boards, `waspmote-pro` has the same `THREAD_STACKSIZE_DEFAULT` and the two other ones are blacklisted.

So with this `4` can be enough to keep for this test.

For information, for `samr21-xpro` I get `4096 (  988)` and `iotlab-m3` I get `4096 ( 1220)`.

**Comment by bergzand** [comment]
@cladmi Sorry for forgetting to change the actual value.

So now change the commit message to reflect the current situation?

**Comment by cladmi** [comment]
By keeping it to `5` for the moment, it may keep other boards we did not re-test still working.
It is the "do not change without reason" solution :)

If you prefer putting 4, just put my stack usage infos as reference.

As long as it is consistent with the change and the justification I am ok.

**Comment by cladmi** [comment]
I "merged" from the web interface to check murdock output.

**Comment by cladmi** [comment]
I am finally back after the release, can you please rebase, I think it was good to go, I will quickly re-test after rebase.

**Comment by bergzand** [comment]
rebased!

**Comment by bergzand** [comment]
I might have screwed something up here with my rebase. I suspect I lost a commit here :(

**Comment by miri64** [comment]
`git reflog` and search ;-). Good luck!

**Comment by cladmi** [comment]
The rebase looks ok for me, by comparing both commits against your rebase base I get this "merge-diff":

```
git show $(git commit-tree 'f9eebce43^{tree}' -p 4631180f1 -p 4eaf1d813 -m commit-tree)

commit 94dca98d9e0be29e3a17edf55093622097ca35eb
Merge: 4631180f1 4eaf1d813
Author: Gaëtan Harter <gaetan.harter@fu-berlin.de>
Date:   Wed Nov 28 19:45:16 2018 +0100

    commit-tree

diff --cc tests/unittests/Makefile
index dfea5bdbf,f24eb63bd..c8dae1960
--- a/tests/unittests/Makefile
+++ b/tests/unittests/Makefile
@@@ -231,8 -231,10 +231,6 @@@ ifneq (,$(filter tests-cpp_%, $(UNIT_TE
    export CPPMIX := 1
  endif
  
 -ifneq (, $(filter $(AVR_BOARDS), $(BOARD)))
 -  LARGE_STACK_TESTS += tests-qDSA
 -endif
--
- LARGE_STACK_TESTS += tests-tweetnacl
  ifneq (,$(filter $(LARGE_STACK_TESTS), $(UNIT_TESTS)))
    CFLAGS += -DTHREAD_STACKSIZE_MAIN=\(4*THREAD_STACKSIZE_DEFAULT+THREAD_EXTRA_STACKSIZE_PRINTF\)
  endif
```

Which corresponds to the rebased version on top of the `tests-tweetnacl` PR.
So for me the rebase is good.

We also note that after this PR, the `LARGE_STACK_TESTS` handling can be removed from `unittests`.

**Pull request merged**